src/main/java/net/mcreator/blockly/java/blocks/EventParameterSetBlock.java

```diff
@@ -0,0 +1,138 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.blockly.java.blocks;
+
+import net.mcreator.blockly.BlocklyBlockUtil;
+import net.mcreator.blockly.BlocklyCompileNote;
+import net.mcreator.blockly.BlocklyToCode;
+import net.mcreator.blockly.IBlockGenerator;
+import net.mcreator.blockly.data.BlocklyLoader;
+import net.mcreator.blockly.data.ExternalTrigger;
+import net.mcreator.blockly.java.BlocklyToProcedure;
+import net.mcreator.generator.GeneratorWrapper;
+import net.mcreator.generator.template.TemplateGeneratorException;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.util.XMLUtil;
+import org.w3c.dom.Element;
+
+import java.util.HashMap;
+import java.util.List;
+
+public class EventParameterSetBlock implements IBlockGenerator {
+
+	@Override public void generateBlock(BlocklyToCode master, Element block) throws TemplateGeneratorException {
+		if (master instanceof BlocklyToProcedure procedure) {
+			if (procedure.getExternalTrigger() == null) {
+				master.addCompileNote(new BlocklyCompileNote(BlocklyCompileNote.Type.ERROR,
+						L10N.t("blockly.errors.event_parameter_set.no_selected_trigger")));
+			} else {
+				ExternalTrigger trigger = null;
+
+				//Try to get the trigger instance for later.
+				List<ExternalTrigger> externalTriggers = BlocklyLoader.INSTANCE.getExternalTriggerLoader()
+						.getExternalTriggers();
+				for (ExternalTrigger externalTrigger : externalTriggers) {
+					if (externalTrigger.getID().equals((procedure.getExternalTrigger()))) {
+						trigger = externalTrigger;
+						break;
+					}
+				}
+
+				//if trigger instance is null
+				if (trigger == null) {
+					master.getCompileNotes().add(new BlocklyCompileNote(BlocklyCompileNote.Type.ERROR,
+							L10N.t("blockly.errors.event_parameter_set.no_selected_trigger")));
+					return;
+				}
+
+				//generator wrapper
+				GeneratorWrapper generatorWrapper = new GeneratorWrapper(procedure.getParent().getGenerator());
+
+				List<Element> elements = XMLUtil.getDirectChildren(block);
+				String value = null, parameter = null, valueType = null;
+				//values
+				for (Element element : elements) {
+					if (element.getNodeName().equals("field")) {
+						String name = element.getAttribute("name");
+						if ("eventparameter".equals(name)) {
+							parameter = element.getTextContent();
+						} else if ("value".equals(name)) {
+							//compatibility with datalist selector, dropdown and so on for later extension
+							value = element.getTextContent();
+							valueType = element.getAttribute("type");
+						}
+					} else if (element.getNodeName().equals("value")) {
+						if (element.getAttribute("name").equals("value")) {
+							value = BlocklyToCode.directProcessOutputBlock(master, element);
+							valueType = BlocklyBlockUtil.getInputBlockType(element);

```

Comment:
Since this comment got lost, I am re-creating it here.

Why is this valueType being queried? It would make sense if you merged procedure blocks into single "set global trigger parameter" block, but currently it does not seem to serve a purpose.

You also already have block type to determine type, and block type locks allow value type anyway.

Please choose one of those:

1. variableType and one single procedure block that accepts any input type and validates it
2. no variableType, but you have multiple procedure blocks